maat-court-data-api/src/main/java/gov/uk/courtdata/dces/service/DebtCollectionRepository.java

```diff
@@ -329,4 +329,25 @@ SELECT COUNT(CC.ID) AS CON_COUNT
         return jdbcTemplate.update(query,delay);
     }
 
+    public List<Integer> getMerge1TestCandidates(int numTestCandidates){
+        // note the double SELECT is to avoid < desired rows being obtained due to ROWNUM and DISTINCT ordering.
+        String sqlStatement = """
+                SELECT * FROM (
+                SELECT DISTINCT RO.ID
+                                FROM TOGDATA.CONCOR_CONTRIBUTIONS CC,
+                                    TOGDATA.REP_ORDERS RO,
+                                    TOGDATA.REP_ORDER_CROWN_COURT_OUTCOMES ROCCO
+                                WHERE CC.REP_ID = RO.ID
+                                AND RO.ID = ROCCO.REP_ID
+                                AND CC.STATUS = 'SENT'
+                                AND ROCCO.CCOO_OUTCOME IS NOT NULL
+                                AND RO.SENTENCE_ORDER_DATE IS NOT NULL
+                                AND TRUNC( ADD_MONTHS( NVL(RO.SENTENCE_ORDER_DATE, SYSDATE ), 5) ) <= TRUNC(SYSDATE)
+                                AND RO.DATE_RECEIVED<TO_DATE('01-JAN-2015')
+                               )
+                               WHERE ROWNUM <= ?
+                """;
+         return jdbcTemplate.queryForList(sqlStatement, Integer.class, numTestCandidates);
```

Comment:
When querying a database connection you have to prepare a statement rather than raw SQL - as part of the OWASP requirements.

Reply:
Double check that's what you're doing here.

Reply:
My understanding for this specific call, is it creates a prepared statement itself. Always happy to be corrected and learn.
I got my info from here: 
https://docs.spring.io/spring-framework/docs/current/javadoc-api/org/springframework/jdbc/core/JdbcTemplate.html#queryForList(java.lang.String,java.lang.Class,java.lang.Object...)

Reply:
Ah perfect, thanks for checking could see a raw String SQL and got a bit nervous!